Lib/multiprocessing/semaphore_tracker.py

```diff
@@ -99,6 +102,17 @@ def ensure_running(self):
             finally:
                 os.close(r)
 
+    def _check_alive(self):
+        '''Check that the pipe has not been closed by sending a probe.'''
+        try:
+            # We cannot use send here as it calls ensure_running, creating
+            # a cycle.
+            os.write(self._fd, b'PROBE:0\n')
+        except BrokenPipeError:
```

Comment:
Perhaps a generic `except OSError` here? That will make `ensure_running` more robust.

Reply:
Ok, I just did the change.